CMakeLists.txt

```diff
@@ -2,11 +2,27 @@
 # Copyright (C) Electronic Arts Inc.  All rights reserved.
 #-------------------------------------------------------------------------------------------
 cmake_minimum_required(VERSION 3.1)
+project(EASTL CXX)
+
+#-------------------------------------------------------------------------------------------
+# Sub-projects
+#-------------------------------------------------------------------------------------------
+add_subdirectory(test/packages/EAAssert)
+add_subdirectory(test/packages/EABase)
+add_subdirectory(test/packages/EAStdC)
+add_subdirectory(test/packages/EAMain)
+add_subdirectory(test/packages/EATest)
+add_subdirectory(test/packages/EAThread)
+
+add_subdirectory(benchmark)
```

Comment:
It's a breaking change to any using the library.  I'd be willing to do that if this is more inline with what most C++ libraries do.  I don't want to be constantly changing the build scripts back and forth.
That being said, that using options would make scripting the build / running of the benchmarks/unit tests would be easier (not require directory hopping).